src/request_template.rs

```diff
@@ -20,6 +20,43 @@ pub struct RequestTemplate {
 }
 
 impl RequestTemplate {
+  fn eval_url2<C: PathString>(&self, ctx: &C) -> anyhow::Result<Url> {
+    let root_url = self.root_url.render(ctx);
+
+    let mut url = url::Url::parse(root_url.as_str())?;
+
+    let q: Vec<(String, String)> = url
+      .query_pairs()
+      .filter_map(|(k, v)| {
+        if v.is_empty() {
+          None
+        } else {
+          Some((k.to_string(), v.to_string()))
+        }
+      })
+      .collect();
```

Comment:
I think we can further optimize by dropping the tuples and collect operations.

Reply:
@tusharmath I believe this has been resolved in the latest commit